src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/hmi_command_factory.cc

```diff
@@ -307,3420 +307,1430 @@ HMICommandFactory::HMICommandFactory(
-                                                          rpc_service_,
-                                                          hmi_capabilities_,
-                                                          policy_handler_));
-      } else {
-        command.reset(new commands::GetSystemInfoRequest(message,
-                                                         application_manager_,
-                                                         rpc_service_,
-                                                         hmi_capabilities_,
-                                                         policy_handler_));
-      }
-      break;
+      return hmi_apis::messageType::request == message_type
+                 ? factory.GetCreator<commands::GetSystemInfoRequest>()
+                 : factory.GetCreator<commands::GetSystemInfoResponse>();
     }
     case hmi_apis::FunctionID::SDL_ActivateApp: {
-      if (is_response) {
-        command.reset(new commands::SDLActivateAppResponse(message,
-                                                           application_manager_,
-                                                           rpc_service_,
-                                                           hmi_capabilities_,
-                                                           policy_handler_));
-      } else {
-        command.reset(new commands::SDLActivateAppRequest(message,
-                                                          application_manager_,
-                                                          rpc_service_,
-                                                          hmi_capabilities_,
-                                                          policy_handler_));
-      }
-      break;
+      return hmi_apis::messageType::request == message_type
+                 ? factory.GetCreator<commands::SDLActivateAppRequest>()
+                 : factory.GetCreator<commands::SDLActivateAppResponse>();
     }
     case hmi_apis::FunctionID::BasicCommunication_PolicyUpdate: {
-      if (is_response) {
-        command.reset(
-            new commands::SDLPolicyUpdateResponse(message,
-                                                  application_manager_,
-                                                  rpc_service_,
-                                                  hmi_capabilities_,
-                                                  policy_handler_));
-      } else {
-        command.reset(new commands::SDLPolicyUpdate(message,
-                                                    application_manager_,
-                                                    rpc_service_,
-                                                    hmi_capabilities_,
-                                                    policy_handler_));
-      }
-      break;
+      return hmi_apis::messageType::request == message_type
+                 ? factory.GetCreator<commands::SDLPolicyUpdateResponse>()
+                 : factory.GetCreator<commands::SDLPolicyUpdate>();
```

Comment:
@BSolonenko 
return hmi_apis::messageType::request == message_type
        command.reset(
                 ? factory.GetCreator<commands::SDLPolicyUpdate>()
            new commands::SDLPolicyUpdateResponse(message,
                : factory.GetCreator<commands::SDLPolicyUpdateResponse>();

Reply:
@LuxoftAKutsan  fixed in 75224d1

Reply:
@LitvinenkoIra   fixed in 75224d1